### DISCUSSION TITLE
How to add custom component pipeline to config file (spacy-nightly)?

### DISCUSSION THREAD

**Discussion by yw2903** [explosion/spaCy#6777] [Help: Coding & Implementations] [answered]
Suppose I define a custom pipeline component using the template and let's call this my_component.py
```
@Language.component("info_component")
def my_component(doc):
```
I can add this component to the config file, pipeline would be ['parser', 'ner', 'my_component'], but how can I link the actual function content (my_component.py) to the config file?

**Comment by svlandeg** [comment]
Hi! Just FYI - we've converted your original issue to a thread on the new Github Discussion forum. We feel this place is more suited for community collaboration & usage discussions, and we're trying to keep the issue tracker more focused on bug reports.

**Comment by adrianeboyd** [comment, marked as answer]
You provide your code with the `--code my_component.py` argument to the relevant CLI commands: `spacy debug`, `spacy init, `spacy train`, etc. See: https://nightly.spacy.io/api/cli

If you're creating a package for your model, you can include your code in the package using the `--code` argument to `spacy package`. ~~I just noticed that this is missing from the docs, so we'll get that updated!~~ [Edited: the docs are fine, I just overlooked it.]

If you're loading a model from a local directory instead of a package, you'll need to make sure your script imports `my_component.py` separately before the model is loaded so that the functions are registered.
